testsuite/tests/typing-jkind-bounds/modalities.ml

```diff
@@ -137,11 +137,11 @@ Line 1, characters 64-65:
 Error: This expression has type "(string -> string) t"
        but an expression was expected of type "('a : value & value)"
        The kind of (string -> string) t is
-         immediate with string -> string @@ global & immediate
-         with string -> string @@ global
+           immediate with string -> string @@ global &
+           immediate with string -> string @@ global
```

Comment:
This is probably outside the scope of this PR, but I think this kind is wrong. iirc, `&` binds tighter than `with`, so this is `(immediate with string -> string @@ global & immediate) with string -> string @@ global`. But what we actually mean is `(immediate with string -> string @@ global) & (immediate with string -> string @@ global)`. (ocamlformat seems to align with my memory.)

Reply:
I've made internal ticket 4547 to track this; well spotted. But I don't want to derail the current PR.